### PR TITLE
Feat/npzt

### PR DESCRIPTION
**Pull request by venkatkapil24**
Allows for cell fluctuations in particular directions.

### CODE REVIEW

**Comment by ceriottm** [comment]
Cannot even run the example in `nickel-npzt-mtk`
```
Per MPI rank memory allocation (min/avg/max) = 3.428 | 3.428 | 3.428 Mbytes
Step Temp KinEng PotEng TotEng Volume Press Density Cella Cellb Cellc CellAlpha CellBeta CellGamma 
       0          273    2.2231452    -284.6877   -282.46455    709.78983   -26272.242    8.7904547    10.012632    10.012632    10.012632           60           60           60 
ERROR: Triclinic box skew is too large (../domain.cpp:216)
Last command: run       100000
```

**Comment by venkatkapil24** [comment]
> Cannot even run the example in `nickel-npzt-mtk`
> 
> ```
> Per MPI rank memory allocation (min/avg/max) = 3.428 | 3.428 | 3.428 Mbytes
> Step Temp KinEng PotEng TotEng Volume Press Density Cella Cellb Cellc CellAlpha CellBeta CellGamma 
>        0          273    2.2231452    -284.6877   -282.46455    709.78983   -26272.242    8.7904547    10.012632    10.012632    10.012632           60           60           60 
> ERROR: Triclinic box skew is too large (../domain.cpp:216)
> Last command: run       100000
> ```

Should work now.

**Comment by ceriottm** [comment]
There's a lot of "whitespace noise" relative to master, as well as several edits that seem completely unrelated to this PR. 
What's going on? No proper linting done on some past PR, or are you reverting linter changes?

**Comment by venkatkapil24** [comment]
> There's a lot of "whitespace noise" relative to master, as well as several edits that seem completely unrelated to this PR.
> What's going on? No proper linting done on some past PR, or are you reverting linter changes?

Well, I did use the i-PI style tool to fix these issues. I assumed that it should lead to the right changes. Which other unrelated edits are you talking about ?

**Comment by ceriottm** [comment]
> > There's a lot of "whitespace noise" relative to master, as well as several edits that seem completely unrelated to this PR.
> > What's going on? No proper linting done on some past PR, or are you reverting linter changes?
> 
> Well, I did use the i-PI style tool to fix these issues. I assumed that it should lead to the right changes. Which other unrelated edits are you talking about ?

So you mean that master is not linter-proof, ATM? I would very much like to focus on more substantial stuff but these things worry me a bit, given all the effort that is going into code clean-up and CI

**Comment by litman90** [comment]
Hi, 
After a  quick check I found that all the white spaces 'diffs' are in docstrings. 

I can't say how this branch created all this modifications in the docstrings (i-pi-style didn't do it), but I can explain why it wasn't detected by our current CI.
In short, black doesn't  check for any specific format inside docstrings and therefore we are not enforcing nothing with flake8
in the workflow.

There are some tools around like https://github.com/myint/docformatter 
It seems reasonable but someone should have a carefull check if it really works ( realeased last Nov) and that it won't bring any other problem.

**Comment by ceriottm** [comment]
@venkatkapil24 can you make changes so all of these pointless whitespace changes go away? it's very distracting to review a PR in which 90% of the diffs are noise.

File: ipi_tests/regression_tests/tests/NPT/gas-npzt-VT/input.xml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+<simulation threading='false'>
+    <ffsocket mode='unix' name='driver'>
```
Comment (eszterpos) [inline ipi_tests/regression_tests/tests/NPT/gas-npzt-VT/input.xml:2]: maybe name = "gas" and address: localhost to follow the conventions of the previous tests?

File: ipi_tests/regression_tests/tests/NPT/gas-npzt-VT/files_to_check.txt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ref_8_simulation.pos_0.xyz xyz
+ref_0_simulation.out numpy
+ref_1_simulation.out numpy
+ref_2_simulation.out numpy
+ref_3_simulation.out numpy
+ref_4_simulation.out numpy
+ref_5_simulation.out numpy
+ref_6_simulation.out numpy
+ref_7_simulation.out numpy
+ref_8_simulation.out numpy
```
Comment (eszterpos) [inline ipi_tests/regression_tests/tests/NPT/gas-npzt-VT/files_to_check.txt:20]: could you format this file so it looks like a table with the header? it works perfectly, it just enhances legibility :) Thanks!

File: ipi_tests/regression_tests/tests/NPT/gas-npzt-VT/driver.txt [hunk 3]
```diff
@@ -0,0 +1,4 @@
+driver gas 
+address gas
```
Comment (eszterpos) [inline ipi_tests/regression_tests/tests/NPT/gas-npzt-VT/driver.txt:2]: address localhost

File: ipi_tests/regression_tests/tests/NPT/gas-npzt-PV/input.xml [hunk 4]
```diff
@@ -0,0 +1,56 @@
+<simulation threading='false'>
+    <ffsocket mode='unix' name='driver'>
+         <latency>  1.00000000e-03</latency>
+         <address> gas </address>
```
Comment (eszterpos) [inline ipi_tests/regression_tests/tests/NPT/gas-npzt-PV/input.xml:4]: just to follow previous examples: address localhost name of ffsocket "gas"?

File: ipi_tests/regression_tests/tests/NPT/gas-npzt-PV/files_to_check.txt [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+ref_0_simulation.out numpy
+ref_1_simulation.out numpy
+ref_2_simulation.out numpy
+ref_3_simulation.out numpy
+ref_4_simulation.out numpy
+ref_0_simulation.pos_0.xyz xyz
+ref_1_simulation.pos_0.xyz xyz
+ref_2_simulation.pos_0.xyz xyz
+ref_3_simulation.pos_0.xyz xyz
+ref_4_simulation.pos_0.xyz xyz
```
Comment (eszterpos) [inline ipi_tests/regression_tests/tests/NPT/gas-npzt-PV/files_to_check.txt:12]: could you format this file so it looks like a table with the header? it works perfectly, it just enhances legibility :) Thanks!

File: ipi_tests/regression_tests/tests/NPT/gas-npzt-PV/driver.txt [hunk 6]
```diff
@@ -0,0 +1,4 @@
+driver gas 
+address gas
```
Comment (eszterpos) [inline ipi_tests/regression_tests/tests/NPT/gas-npzt-PV/driver.txt:2]: address localhost?

**Review by eszterpos — COMMENTED** [review]
My comments are regarding simple formatting things, the regression tests are running fine and their setup is mostly following our previous existing regression tests. Even if you decide not to address the comments, it can be merged :)

**Comment by venkatkapil24** [comment]
Looks like its all good @eszterpos. If you are satisfied please approve the PR :)

**Pull request merged**